kafka-client-examples/e2e-test/src/main/java/dev/responsive/examples/common/UncaughtStreamsAntithesisHandler.java

```diff
@@ -77,12 +77,10 @@ private boolean shouldLogError(final Throwable throwable, List<Throwable> seen)
         InjectedE2ETestException.class,
         InvalidProducerEpochException.class,
         MongoNotPrimaryException.class,
+        MongoTimeoutException.class,
         ProducerFencedException.class,
         ReadFailureException.class,
         ReadTimeoutException.class,
-        RebalanceInProgressException.class,
-        TaskCorruptedException.class,
-        TaskMigratedException.class,
```

Comment:
Sanity check me here I guess, but none of these should ever get thrown all the way up to the uncaught exception handler in Streams. So if we do ever see something like these it would be a bug that we should look into (whether in Responsive code or AK)

Reply:
cc @rodesai - I believe you added these? What was the reason for that?

Reply:
I took them all to be things that you'd expect to see in the presence of faults.

I may have misunderstood `RebalanceInProgressException` - I thought you'd hit that if the group hits a rebalance while the application is processing and it calls something other than `poll`.

From what I see `TaskCorruptedException` can be thrown for various transient errors like the broker not being available or if `process` hits a timeout. Ideally we can differentiate these from the state actually being corrupted, but all the cases I saw were transient errors.

`TaskMigratedException` seems pretty expected right? Streams seems to throw it whenever it detects the producer's been fenced. And it gets thrown when by us when we fail to reserve an epoch because the task got fenced.

Reply:
Ohhhh I see what you're saying - streams should catch these and handle them transparently. That makes sense to me. Lets drop them and see what happens.

Reply:
yeah, these are "expected" exceptions but they should never be thrown all the way up to the user's exception handler, and should be handled by Streams